src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,17 +1,121 @@
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.IBookRepository;
 
-@Controller
+
+/**
+ * This class is the service responsible of handling the requests related to the book api
+ */
+
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private IBookRepository bookRepository;
+
     @GetMapping("/greeting")
-    public String greeting(@RequestParam(name="name", required=false, defaultValue="Poli") String name, Model model) {
+    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "Poli") String name,
+            Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
 
+    /**
+     * @return all the books contained in the model
+     */
+    @GetMapping("/all")
+    public Optional<List<Book>> findAll() {
+        List<Book> allBooks = new ArrayList<>();
+        if (bookRepository.findAll().iterator().hasNext()) {
+            bookRepository.findAll().forEach(b -> allBooks.add(b));
+            return Optional.of(allBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * @param bookAuthor indicates the author to be searched
+     *
+     * @return the entire book of the selected author
+     */
+    @GetMapping("/author/{bookAuthor}")
+    public Optional<List<Book>> findByAuthor(@PathVariable String bookAuthor) {
+        List<Book> authorBooks = new ArrayList<>();
+        if (bookRepository.findByBookAuthor(bookAuthor).iterator().hasNext()) {
+            bookRepository.findByBookAuthor(bookAuthor).forEach(b -> authorBooks.add(b));
+            return Optional.of(authorBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @GetMapping("/{bookGenre}")
+    public Optional<List<Book>> findByGenre(@PathVariable String bookGenre) {
+        List<Book> genreBooks = new ArrayList<>();
+        if (bookRepository.findByBookGenre(bookGenre).iterator().hasNext()) {
+            bookRepository.findByBookGenre(bookGenre).forEach(b -> genreBooks.add(b));
+            return Optional.of(genreBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * @param book indicates the book to be added
+     *
+     * @return the created book
+     */
+    @PostMapping("/create")
```

Comment:
Like the `findAll` resources, for creating a new item we should use the resource's root path. Furthermore, the best practices API indicates that verbs shouldn't be used.
```suggestion
    @PostMapping
```